src/background/requests/registerChain.ts

```diff
@@ -0,0 +1,41 @@
+import { ZerionAPI } from 'src/modules/zerion-api/zerion-api';
+import {
+  getBackendNetworkByChainId,
+  getBackendNetworkByLocalChain,
+} from 'src/modules/networks/getBackendNetwork';
+import { normalizeAddress } from 'src/shared/normalizeAddress';
+import { emitter } from '../events';
+import type { Account } from '../account/Account';
+
+export function initialize(account: Account) {
+  function listenForChainChange() {
+    account.getCurrentWallet().emitter.on('chainChanged', async (chain) => {
+      const network = await getBackendNetworkByLocalChain(chain);
+      const address = account.getCurrentWallet().readCurrentAddress();
+      if (network && address) {
+        ZerionAPI.registerChain({
+          chain: network.id,
+          addresses: [normalizeAddress(address)],
+        });
+      }
+    });
+  }
+  listenForChainChange();
+  // Update subscription after Wallet's recreation inside Account
+  account.on('reset', () => {
+    listenForChainChange();
+  });
+
```

Comment:
Currently events aren't cleaned up

This re-application of listeners might not be needed if we use the global emitter

Reply:
Yes. I didn't clean them because this reset event means that emitter was destroyed anyway.
But I will reuse global emitter anyway